RecoEgamma/EgammaTools/plugins/HGCalElectronFilter.cc

```diff
@@ -0,0 +1,180 @@
+
+    if (nElectrons > 0 ) {
+        for(unsigned iEle1 = 0; iEle1 < nElectrons; ++iEle1) {
+            bool isBest = true;
+            const auto& electron1 = ElectronsH->at(iEle1);
+            if (!cleanBarrel_ && electron1.isEB()) {// keep all barrel electrons
+                isBest = true;
+                gsfElectrons_p->push_back(electron1);
+                continue;
+            } else {
+                for (unsigned iEle2 = 0; iEle2 < nElectrons; ++iEle2) {
+                    if (iEle1 == iEle2) continue;
+                    const auto& electron2 = ElectronsH->at(iEle2);
+                    if (electron1.superCluster() != electron2.superCluster()) continue;
+                    if (electron1.electronCluster() != electron2.electronCluster()) {
+                        if (electron1.electronCluster()->energy() < electron2.electronCluster()->energy()) {
+                            isBest=false;
+                            break;
+                        }
+                    } else {
+                        if (fabs(electron1.eEleClusterOverPout()-1.) > fabs(electron2.eEleClusterOverPout()-1.)) {
+                            isBest=false;
+                            break;
+                        }
+                    }
+                }
+                if (isBest) gsfElectrons_p->push_back(electron1);
+            }
+        }
+    }
+
+    iEvent.put(std::move(gsfElectrons_p),outputCollection_);
+}
+
+// ------------ method called once each stream before processing any runs, lumis or events  ------------
+void
+HGCalElectronFilter::beginStream(edm::StreamID)
+{
+}
+
+// ------------ method called once each stream after processing all runs, lumis and events  ------------
+void
+HGCalElectronFilter::endStream() {
+}
+
+// ------------ method called when starting to processes a run  ------------
+/*
+void
+HGCalElectronFilter::beginRun(edm::Run const&, edm::EventSetup const&)
+{
+}
+*/
+
+// ------------ method called when ending the processing of a run  ------------
+/*
+void
+HGCalElectronFilter::endRun(edm::Run const&, edm::EventSetup const&)
+{
+}
+*/
+
+// ------------ method called when starting to processes a luminosity block  ------------
+/*
+void
+HGCalElectronFilter::beginLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&)
+{
+}
+*/
+
+// ------------ method called when ending the processing of a luminosity block  ------------
+/*
+void
+HGCalElectronFilter::endLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&)
+{
+}
+*/
+
+// ------------ method fills 'descriptions' with the allowed parameters for the module  ------------
+void
+HGCalElectronFilter::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  //The following says we do not know what parameters are allowed so do no validation
+  // Please change this to state exactly what you do use, even if it is no parameters
+  edm::ParameterSetDescription desc;
+  desc.setUnknown();
+  descriptions.addDefault(desc);
```

Comment:
please implement non-empty fillDescriptions and create hgcalElectronFilter config instance here.
After that the hgcalElectronFilter_cfi.py file can be removed.